File: lesson14-kotlin-type-system/ElapsedTimeCalculator.java
public class ElapsedTimeCalculator {
    // The singleton instance
    public static final ElapsedTimeCalculator INSTANCE;

    static {
        INSTANCE = new ElapsedTimeCalculator();
    }

    private static final long startTime = 
        System.currentTimeMillis();

    // Avoid the instantiation of this class
    // out of the scope of this class.
    private ElapsedTimeCalculator() {
    }
    public long elapsedTime() {
        long currentTime = System.currentTimeMillis();
        return currentTime - startTime;
    }
}

File: lesson14-kotlin-type-system/ElapsedTimeCalculator.kt
object ElapsedTimeCalculator {
    private val startTime: Long= System.currentTimeMillis()

    fun elapsedTime(): Long {
        val currentTime = System.currentTimeMillis()
        return currentTime - startTime
    }
}

fun main(args: Array<String>) {
    println(ElapsedTimeCalculator.elapsedTime()) // <=> ElapsedTimeCalculator.INSTANCE.elapsedTime()
    Thread.sleep(2500)
    println(ElapsedTimeCalculator.elapsedTime())
    
}